Mle pca implementation

CeeThinwa opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Fixes the PCA implementation does not match Tipping and Bishop #10137 issue

#### What does this implement/fix? Explain your changes.
The explained variance formula did not match Tipping and Bishop eq (11) so this was changed to match the reference material.

#### Any other comments?
We submitted this in a team of 2; @CeeThinwa and @fortune-uwha for the AFME sklearn sprint

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

CeeThinwa committed:
Merge pull request #2 from scikit-learn/main

Update to forked repo

CeeThinwa committed:
Fixed issue where explained variance formula was changed to match Tipping and Bishop eq (11)

CeeThinwa committed:
Fixed issue where explained variance formula was changed to match Tipping and Bishop eq (11)

reshamas commented:
#DataUmbrella sprint

lorentzenchr commented:
I think the recommendation of https://github.com/scikit-learn/scikit-learn/issues/10137#issuecomment-344733245 is not to change the denominator from `n_sampels - 1` to `n_samples` (the `-1` was introduced in #9105), but to chance the docsting.

CeeThinwa commented:
Thank you @lorentzenchr for your remarks; I will change the docstring and modify the pull request to mention the same.

CeeThinwa committed:
Changed the docstring not the actual formula

Changed the docstring under Tipping and Bishop to show their understanding of explained variance, but changed explained variance back to the original
`explained_variance_ = (S ** 2) / (n_samples-1)`

lorentzenchr commented on sklearn/decomposition/_pca.py:
```diff
@@ -283,6 +283,8 @@ class PCA(_BasePCA):
     Series B (Statistical Methodology), 61(3), 611-622.
     <http://www.miketipping.com/papers/met-mppca.pdf>`_
     via the score and score_samples methods.
+    The explained variance according to the two is
+    explained_variance_ = (S ** 2) / n_samples
```
The right place to put a comment is above, under the attribute section. There, one should just say that `n_samples - 1` degrees of freedoms are used for the variance estimation.

CeeThinwa committed:
Update _pca.py

lorentzenchr commented on sklearn/decomposition/_pca.py:
```diff
@@ -217,6 +217,7 @@ class PCA(_BasePCA):
 
     explained_variance_ : ndarray of shape (n_components,)
         The amount of variance explained by each of the selected components.
+        The degrees of freedom for variance estimation are (n_samples - 1)
```
```suggestion
        The variance estimation uses `n_samples - 1` degrees of freedom.
```

CeeThinwa committed:
Update sklearn/decomposition/_pca.py documentation

Co-authored-by: Christian Lorentzen <lorentzen.ch@gmail.com>

CeeThinwa commented:
@lorentzenchr just committed your suggestion. Thanks for your guidance.

lorentzenchr reviewed (approved):
LGTM

lorentzenchr commented:
@CeeThinwa Thank you for fixing this doc issue.

glemaitre commented:
Thank you @CeeThinwa and @fortune-uwha
